controllers/cloud.redhat.com/providers/database/localdb.go

```diff
@@ -106,7 +106,14 @@ func (db *localDbProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 	}
 
 	labels := &map[string]string{"sub": "local_db"}
-	provutils.MakeLocalDB(dd, nn, app, labels, &dbCfg, image, db.Env.Spec.Providers.Database.PVC, app.Spec.Database.Name, nil)
+
+	databaseResourceRequirements := app.Spec.Database.Resources
+
+	if databaseResourceRequirements.Limits == nil || databaseResourceRequirements.Requests == nil {
```

Comment:
I wonder if we should instead be checking if `app.Spec.Database.Resources` is "empty". That would look like:
```
emptyResourceRequirements := v1.ResourceRequirements{}
if databaseResourceRequirements == emptyResourceRequirements  {
    // use defaults
}
```

Reason is, it is valid to define a resource block which sets no value for requests but DOES set limits (it may rarely be used that way but if we want it to be truly a 'pass-thru' config we'd probably want to consider that).

Reply:
I don't know what `v1` is here. Its not in scope. There are a couple of packages we import that have `v1` in their paths like `apps "k8s.io/api/apps/v1"` and  `core "k8s.io/api/core/v1"` but neither of them expose a `ResourceRequirements` - where should I be pulling that from?

Reply:
NVM - its core. VSCode's completion just wasn't catching up.

Reply:
Here's my code btw

```
	emptyResourceRequirements := core.ResourceRequirements{}
	databaseResourceRequirements := app.Spec.Database.Resources

	if databaseResourceRequirements == emptyResourceRequirements {
		databaseResourceRequirements = provutils.DatabaseResourceDefaults()
	}
```

Reply:
Should this work? Internet seems to say that our options for struct comparisons are either slow (reflect.DeepCopy), dangerous (cmp.Equal), or roll your own (check equality of struct members.) I have no idea if that's true or not.

Reply:
@bsquizz let me know what you think re: the above. If we can't just compare the structs directly that might be why we did the initial code checking for the members being null.